20241014-wordcloud/src/routes/WordCloud.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  import { createEventDispatcher } from "svelte";
+  import cloud from "d3-cloud";
+  import { select } from "d3-selection";
+  import { scaleOrdinal } from "d3-scale";
+  import * as CS from "d3-scale-chromatic";
+
+  // event dispatcher
+  const dispatch = createEventDispatcher();
+
+  // color scheme
+  const color_scheme = {
+    schemeCategory10: CS.schemeCategory10,
+    schemeAccent: CS.schemeAccent,
+    schemeDark2: CS.schemeDark2,
+    schemePaired: CS.schemePaired,
+    schemePastel1: CS.schemePastel1,
+    schemePastel2: CS.schemePastel2,
+    schemeSet1: CS.schemeSet1,
+    schemeSet2: CS.schemeSet2,
+    schemeSet3: CS.schemeSet3,
+    schemeTableau10: CS.schemeTableau10,
+  };
+
+  // props
+  export let words;
+  export let width;
+  export let height;
+  export let font = "Fira Sans";
+  // Fudge factor to make it more likely for all words to fit
+  export let maxFontSize = Math.floor(height / words.length);
+  export let minRotate = 0;
+  export let maxRotate = 0;
+  export let scheme = "schemePaired";
+  export let padding = 2;
+  export let backgroundColor = "#fff"
+
+  // count max word occurence
+  const maxWordCount = words.reduce((prev, cur) =>
+    prev.count < cur.count ? prev.count : cur.count
+  );
```

Comment:
```suggestion
  var maxWordCount = words.reduce((prev, cur) =>
    prev > cur.count ? prev : cur.count, 0
  );
```
The accumulator takes the same type as the return value, so here `prev` is of type `number` whereas `cur` is of type `Object`, so this needs a slightly different formulation. I think your inequality needs reversing as well. You need an initialiser to avoid it returning an `Object` in case there's only one item in the array.

I only noticed because this was returning strange values; I had to look all this stuff up of course.

This avoids the situation where the wordcloud font gets tinier if you repeatedly enter the same value. But having said all that, I actually found I got much better results if `maxWordCount` was always set to 1.